prefect_aws/ecs.py

```diff
@@ -0,0 +1,981 @@
+            and not values.get("execution_role_arn")
+            # Do not raise if they've linked to another task definition or provided
+            # it without using our shortcuts
+            and not values.get("task_definition_arn")
+            and not (values.get("task_definition") or {}).get("executionRoleArn")
+        ):
+            raise ValueError(
+                "An `execution_role_arn` must be provided to use "
+                "`configure_cloudwatch_logs` or `stream_logs`."
+            )
+        return values
+
+    @root_validator
+    def image_is_required(cls, values: dict) -> dict:
+        """
+        Enforces that an image is available if the user sets it to `None`.
+        """
+        if (
+            not values.get("image")
+            and not values.get("task_definition_arn")
+            # Check for an image in the task definition; whew!
+            and not (
+                get_prefect_container(
+                    (values.get("task_definition") or {}).get(
+                        "containerDefinitions", []
+                    )
+                )
+                or {}
+            ).get("image")
+        ):
+            raise ValueError(
+                "A value for the `image` field must be provided unless already "
+                "present for the Prefect container definition a given task definition."
+            )
+        return values
+
+    @sync_compatible
+    async def run(self, task_status: Optional[TaskStatus] = None) -> ECSTaskResult:
+        """
+        Run the configured task on ECS.
+        """
+        boto_session, ecs_client = await run_sync_in_worker_thread(
+            self._get_session_and_client
+        )
+
+        (
+            task_arn,
+            cluster_arn,
+            task_definition,
+            is_new_task_definition,
+        ) = await run_sync_in_worker_thread(
+            self._create_task_and_wait_for_start, boto_session, ecs_client
+        )
+
+        # Display a nice message indicating the command and image
+        self.logger.info(
+            f"{self.log_prefix}: Running command {' '.join(self.command)!r} "
+            f"in container {PREFECT_ECS_CONTAINER_NAME!r} ({self.image})..."
+        )
+
+        if task_status:
+            task_status.started(task_arn)
+
+        status_code = await run_sync_in_worker_thread(
+            self._watch_task_and_get_exit_code,
+            task_arn,
+            cluster_arn,
+            task_definition,
+            is_new_task_definition and self.auto_deregister_task_definition,
+            boto_session,
+            ecs_client,
+        )
+
+        return ECSTaskResult(
+            identifier=task_arn,
+            # If the container does not start the exit code can be null but we must
+            # still report a status code. We use a -1 to indicate a special code.
+            status_code=status_code or -1,
+        )
+
+    @property
+    def log_prefix(self) -> str:
```

Comment:
If this is intertal, should it be named `_log_prefix`? It mostly depends on if you want it to show in the docs or not.